core/src/main/java/org/infinispan/configuration/cache/Configuration.java

```diff
@@ -303,4 +319,13 @@ public String toStringConfiguration(String name, MediaType mediaType, boolean cl
       }
       return sw.toString();
    }
+
+   public static final class AliasMatcher implements AttributeMatcher<List<String>> {
+      static final AliasMatcher INSTANCE = new AliasMatcher();
+
+      @Override
+      public boolean matches(Attribute<List<String>> o1, Attribute<List<String>> o2) {
+         return true;
```

Comment:
what's its purpose if it returns always true?

Reply:
I have reworked this a little to provide a better explanation why we're doing this